install/0000_90_machine-config-operator_01_prometheus-rules.yaml

```diff
@@ -57,3 +57,23 @@ spec:
             severity: warning
           annotations:
             message: "System memory usage of {{ $value | humanize }} on {{ $labels.node }} exceeds 90% of the reservation. Reserved memory ensures system processes can function even when the node is fully allocated and protects against workload out of memory events impacting the proper functioning of the node. The default reservation is expected to be sufficient for most configurations and should be increased (https://docs.openshift.com/container-platform/latest/nodes/nodes/nodes-nodes-managing.html) when running nodes with high numbers of pods (either due to rate of change or at steady state)."
+    - name: containers-oom-killed-warning
+      rules:
+        - alert: ContainersOOMKilledWarning
+          expr: |
+            sum by (node) (container_runtime_crio_containers_oom_total) > 1
+          for: 15m
+          labels:
+            severity: warning
+          annotations:
+            message: "{{ $value | humanize }} containers have been killed on {{ $labels.node }} because they ran out of memory (OOM). Please make sure to limit the memory for the containers to not overcommit the node or try to reduce the total number of pods per node."
+    - name: containers-oom-killed-critical
+      rules:
+        - alert: ContainersOOMKilledCritical
+          expr: |
+            sum by (node) (container_runtime_crio_containers_oom_total) > 5
```

Comment:
As I understand it, the criteria here is to alert on the total OOM killed pods at a node level crossing threshold within 15 minutes. I have following thoughts/queries:
1. Correct me if am wrong but I think the count of this metric will also increase if an insufficient memory limit is set for any app. In a scenario where the node has enough free memory but end user/customer app's (any app not in openshift-* or kube-* namespace) memory limit is insufficient, this OOM killed count will still increase and the rate can increase with more apps. For SRE, this alert would not be actionable at all if it's purely customer app configuration issue. 
2. Regarding OOM killed containers in general, if we as such scope it for the node then do you think it could be better to refer to the free/available node memory as the eventual symptom to alert upon? Else can scope the OOM killed alert for core openshift-*, kube-* and default namespaces?
3. If anyone gets paged for this alert, the message indicates that "some containers are OOM killed on a node". This could be any container/pod across all the namespaces. I think with reference to above two points, we could make the "actionable" message to be more informative.

Reply:
@ravitri Thanks for your input. I am happy to take any suggestions to make the alert message more actionable.